backend/molgenis-emx2-rdf/src/test/java/org/molgenis/emx2/rdf/InMemoryRDFHandler.java

```diff
@@ -0,0 +1,47 @@
+package org.molgenis.emx2.rdf;
+
+import java.util.*;
+import org.eclipse.rdf4j.model.*;
+import org.eclipse.rdf4j.model.util.Values;
+import org.eclipse.rdf4j.rio.RDFHandler;
+import org.eclipse.rdf4j.rio.RDFHandlerException;
+
+abstract class InMemoryRDFHandler implements RDFHandler {
+  public Map<Resource, Map<IRI, List<Value>>> resources = new HashMap<>();
+  public List<Namespace> namespaces = new ArrayList<>();
+
+  @Override
+  public void handleNamespace(String prefix, String uri) throws RDFHandlerException {
+    namespaces.add(Values.namespace(prefix, uri));
+  }
+
+  @Override
+  public void handleStatement(Statement st) throws RDFHandlerException {
+    resources.merge(
```

Comment:
a line comment on what this does would help me undertand

Reply:
I looked at the code again and rewrote it using Map::getOrDefault calls which made the code much simpler.